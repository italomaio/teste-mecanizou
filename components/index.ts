export { default as Card } from "./Card/Card";
export { default as RootProviders } from "./RootProviders/RootProviders";
export { default as Logo, type LogoProps } from "./Logo/Logo";
export { default as Input } from "./Input/Input";
export { default as Label, type LabelProps } from "./Label/Label";
export {
  default as Button,
  type ButtonVariant,
  type ButtonSize,
  type ButtonProps,
} from "./Button/Button";
export { default as ThemeToggle } from "./ThemeToggle/ThemeToggle";
export { default as ProductLink } from "./ProductLink/ProductLink";
export { default as Navbar } from "./Navbar/Navbar";
export { default as Skeleton } from "./Skeleton/Skeleton";
export { default as Pagination } from "./Pagination/Pagination";
export { default as ProductRating } from "./ProductRating/ProductRating";
export { default as ProductCard } from "./ProductCard/ProductCard";
export { default as ProductQuantity } from "./ProductQuantity/ProductQuantity";
export { default as ProductImages } from "./ProductImages/ProductImages";
export { default as ProductReviews } from "./ProductReviews/ProductReviews";
export { default as Stars } from "./ProductRating/Stars";
export { default as Loader } from "./Loader/Loader";
export { default as ProductPrices } from "./ProductPrices/ProductPrices";
export {
  default as UserDropdown,
  type UserDropdownProps,
} from "./UserDropdown/UserDropdown";
