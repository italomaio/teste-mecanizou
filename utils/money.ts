export const formatter = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
  minimumFractionDigits: 2,
});

export function calculateDiscountPrice(
  price: number,
  discountPercentage: number
) {
  const discountFactor = discountPercentage / 100;
  const newPrice = price * (1 - discountFactor);

  return {
    oldPrice: formatter.format(price),
    newPrice: formatter.format(newPrice),
  };
}
