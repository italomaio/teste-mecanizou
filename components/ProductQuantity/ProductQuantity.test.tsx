import { render, fireEvent, screen } from "@/tests/utils";
import { ProductQuantity } from "@/components";

describe("ProductQuantity", () => {
  it("Should render correctly", () => {
    const { container } = render(<ProductQuantity min={10} />);
    expect(container).toBeInTheDocument();
  });

  it("Should call onChange", () => {
    const onChangeMock = jest.fn();
    render(<ProductQuantity min={10} onChangeValue={onChangeMock} />);
    fireEvent.change(screen.getByLabelText(/Quantity/i), {
      target: { value: 1 },
    });
    expect(onChangeMock).toHaveBeenCalled();
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(<ProductQuantity min={10} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
