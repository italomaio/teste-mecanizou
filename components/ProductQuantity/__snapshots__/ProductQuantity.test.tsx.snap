// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductQuantity Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-1/6 flex flex-col mb-[2px]"
  >
    <label
      class="inline-block focus-visible-ring p-0 mb-2 text-left font-medium text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-gray-800 dark:text-white"
      for="qty"
    >
      Quantidade:
    </label>
    <input
      aria-label="Quantity"
      aria-required="true"
      class="flex h-9 w-full rounded-sm bg-transparent px-3 py-1 text-base shadow-sm transition-colors placeholder:text-muted-foreground disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-sapphire text-gray-700 dark:text-zinc-100 border border-sapphire file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground"
      id="qty"
      min="10"
      role="textbox"
      type="number"
      value="10"
    />
  </div>
</DocumentFragment>
`;
